lib/resources/dstu2/medication_statement.yaml

```diff
@@ -0,0 +1,259 @@
+                     "unit": "tabs"}}}
+    url: http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.dosage.quantity_x_
+
+- name: patient
+  required: 'Yes'
+  cardinality: 1..1
+  type: Reference (Patient)
+  description: Who is/was taking the medication.
+  example: '{"patient": {"reference": "Patient/5366327"}}'
+
+- name: reasonForUse[x]
+  required: 'No'
+  cardinality: 0..1
+  type: CodeableConcept|Reference(Condition)
+  description: A reason for why the medication is being/was taken.
+  example: |
+             reasonForUseReference
+               {"reasonForUseReference": {
+                 "reference": "Condition/123",
+                 "display": "Acute sinusitis"}}
+
+             reasonForUseCodeableConcept
+               {"reasonForUseCodeableConcept": {
+                 "coding": [
+                   {"system": "http://snomed.info/sct",
+                    "code": "25064002",
+                    "display": "Otitis Media"}]}}
+  url: http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.reasonForUse_x_
+  binding:
+    description: Codes identifying why the medication is being taken.
+    terminology:
+    - display: SNOMED CT
+      system: http://snomed.info/sct
+
+- name: status
+  required: 'Yes'
+  cardinality: 1..1
+  type: code
+  description: >
+                 A code representing the patient or other source's judgment about the state of the medication
+                 used that this statement is about.
+  example: '{"status": "active"}'
+
+- name: note
+  required: 'No'
+  cardinality: 0..1
+  type: string
+  description: Provides extra information about the medication statement that is not conveyed by the other attributes.
+  example: '{"note": "do not take with alcohol"}'
+
+- name: effective[x]
+  required: 'No'
+  cardinality: 0..1
+  type: dateTime|Period
+  description: The interval of time during which it is being asserted that the patient was taking the medication.
+  example: |
+             effectiveDateTime
+               {"effectiveDateTime": "2015-05-05T14:00:00.000Z"}
+
+             effectivePeriod
+               {"effectivePeriod": {
+                 "start": "2015-05-05T14:00:00.000Z",
+                 "end": "2015-05-06T01:00:00.000Z"}}
+  url: http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.effective_x_
+
+- name: informationSource
+  required: 'No'
+  cardinality: 0..1
+  type: Reference(Patient | Practitioner | RelatedPerson)
+  description: 	The person who provided the information about the taking of this medication.
+  example: |
+             {"informationSource": {
+               "reference": "Practitioner/3150783",
+               "display": "John Smith"}}
+
+- name: medication[x]
+  required: 'Yes'
+  cardinality: 1..1
+  type: CodeableConcept|Reference(Medication)
+  description: Identifies the medication being administered.
+  example: |
+             medicationCodeableConcept
```

Comment:
Do we support medicationCodeableConcept on create? I thought we currently required a contained reference?